protocol/test/temple-zaps.ts

```diff
@@ -0,0 +1,2056 @@
+      pair.address,
+      zapInFor,
+      ZEROEX_EXCHANGE_PROXY,
+      swapCallData,
+      zapLiqRequest,
+      overrides
+    ))
+    .to.emit(zaps, "ZappedLPUniV2");
+  }
+
+  // checks
+  const liquidityAfter = await pair.balanceOf(zapInFor);
+  const signerTokenBalanceAfter = await getBalance(fromTokenContract, signerAddress);
+  expect(liquidityAfter.sub(liquidityBefore)).to.gt(0);
+
+  // ensure no residual tokens
+  if (shouldTransferResidual) {
+    expect(signerTokenBalanceBefore.sub(signerTokenBalanceAfter)).to.lte(fromAmountBN);
+
+    const zapsTokenBalanceAfter = await getBalance(fromTokenContract, zaps.address);
+    const zapsPairToken0BalanceAfter = await getBalance(pairToken0Contract, zaps.address);
+    const zapsPairToken1BalanceAfter =await getBalance(pairToken1Contract, zaps.address);
+    const pairToken0BalanceAfter = await getBalance(pairToken0Contract, zapInFor); 
+    const pairToken1BalanceAfter = await getBalance(pairToken1Contract, zapInFor);
+    expect(zapsTokenBalanceAfter.sub(zapsTokenBalanceBefore)).to.eq(0);
+    expect(zapsPairToken0BalanceAfter.sub(zapsPairToken0BalanceBefore)).to.eq(0);
+    expect(zapsPairToken1BalanceAfter).to.eq(zapsPairToken1BalanceBefore);
+    // user residuals received
+    expect(pairToken0BalanceAfter).to.gte(pairToken0BalanceBefore);
+    expect(pairToken1BalanceAfter).to.gte(pairToken1BalanceBefore);
+  } else {
+    expect(signerTokenBalanceBefore.sub(signerTokenBalanceAfter)).to.gte(fromAmountBN);
+  }
+}
+
+async function zapInTempleFaith(
+  signer: Signer,
+  zapInfor: string,
+  fundAmount: BigNumber,
+  fromAmount: string,
+  fromTokenAddress: string,
+  fee: BigNumber
+) {
+  const tokenContract = IERC20__factory.connect(fromTokenAddress, signer);
+  let symbol;
+  let decimals;
+  let sellToken;
+  if (fromTokenAddress === ETH) {
+    symbol = 'ETH';
+    decimals = 18;
+    sellToken = 'ETH';
+  } else {
+    symbol = await tokenContract.symbol();
+    decimals = await tokenContract.decimals();
+    sellToken = fromTokenAddress;
+  }
+  // get some faith for signer
+  const signerAddress = await signer.getAddress();
+  const faithAmount = 10000;
+  await faith.gain(signerAddress, faithAmount);
+  const signerFaithBalance = await faith.balances(signerAddress);
+  expect(signerFaithBalance.usableFaith).to.eq(faithAmount);
+
+  // zap temple+faith
+  const fromToken = ERC20__factory.connect(fromTokenAddress, signer);
+  const templeToken = IERC20__factory.connect(TEMPLE, signer);
+
+  // Get TEMPLE balance before zap
+  const balanceBefore = signerAddress == zapInfor ? await getBalance(templeToken, signerAddress) : await getBalance(templeToken, zapInfor);
+  console.log(
+    `Starting Temple: ${ethers.utils.formatUnits(balanceBefore, 18)}`
+  );
+  console.log(`Selling ${fromAmount} ${symbol}`);
+
+  // Approve token
+  if (fromTokenAddress !== ETH) {
+    await tokenContract.approve(
+      templeZaps.address,
+      ethers.utils.parseUnits('1000111', decimals)
```

Comment:
Meant to remain as this odd amount? Are you intending the `console.log()` to remain?

Reply:
the initial intention was to leave the console logs only for the price info.
@scooprinder you also got any views on this?